parlai/core/logging_utils.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import sys
+import logging
+from pathlib import Path
+from datetime import date
+
+INFO_LEVEL = logging.INFO
+DEBUG_LEVEL = logging.DEBUG
+WARN_LEVEL = logging.WARNING
+ERROR_LEVEL = logging.ERROR
+CRITICAL_LEVEL = logging.CRITICAL
+
+
+class ParlaiLogger(logging.Logger):
+    def __init__(
+        self,
+        name,
+        console_level,
+        console_format=None,
+        file_format=None,
+        file_level=logging.INFO,
+        filename=None,
+    ):
+        """
+        Initialize the logger object
+        :param name: Name of the logger
+        :param console_level: min. Level of messages logged to console
+        :param console_format: The format of messages logged to the console. Simple stdout is used if None specified
+        :param file_format: The format of messages logged to the file (A default is used if None specified)
+        :param file_level: min. Level of messages logged to the file
+        :param filename: The file the logs are written to
+
+        A default file is initialized if no filename was provided. It can be
+        found in ParlAI/Logs/*year*/*month*/*YYYY-MM-DD*.log
+        """
+        # Setup logging location if none specified
+        if filename is None:
```

Comment:
I'm not sure I'm ready for default logging-to-file always. This will clobber the NFS we use and anger our admins. Let's just have if filename is None do nothing, and you can document where the right way place to do this is.

Additionally I run probably dozens of commands every day, they'd all overwrite each other without a microsecond stamp.